### ISSUE TITLE
Variable inputs not detecting in serverless.yml

### ISSUE THREAD

**Issue by kueckermann** [serverless/components#763]
# Description
When trying to use ${...} to input dynamic variables into my serverless.yml it does not parse them.

## Additional Data
Using `stage: ${opt:stage, 'dev'}` in my yml file results in serverless setting my stage name to **"${opt:stage, 'dev'}"**

E.g. stack trace, version, operating system, ...
Operating System:          darwin
Node Version:              12.16.3
Framework Version:         1.78.1
Plugin Version:            3.7.0
SDK Version:               2.3.1
Components Version:        2.34.3

**Comment by eahefnawy** [comment]
yeah, only the variables listed here are supported when using components:
https://github.com/serverless/components#variables

Everything else is only supported when using the "traditional" serverless framework.

We actually have error handling that should give you this feedback quickly, but it seems there's a bug there when using fallback `${opt:stage, dev}`... we will fix that 😅
